app/controllers/spree/api/products_controller_decorator.rb

```diff
@@ -89,12 +89,16 @@ def render_paged_products(products)
 
     render text: {
       products: serializer,
-      pagination: {
-        results: products.total_count,
-        pages: products.num_pages,
-        page: params[:page].to_i,
-        per_page: params[:per_page].to_i
-      }
+      pagination: pagination_data(products)
     }.to_json
   end
+
+  def pagination_data(results)
+    {
+      results: results.total_count,
+      pages: results.num_pages,
+      page: params[:page].to_i,
+      per_page: params[:per_page].to_i
+    }
```

Comment:
I'd refer to see json-api structure here so we get our API pagination exemplar done:
https://jsonapi.org/examples/#pagination


```ruby
  meta: {
    totalPages: results.total_count
  },
  data: serializer,
  links: {
    self: "/api/products?page[number]=#{params[:page].to_i}&page[size]=1",
    first: "http://example.com/articles?page[number]=1&page[size]=1",
     prev: "http://example.com/articles?page[number]=2&page[size]=1",
     next: "http://example.com/articles?page[number]=4&page[size]=1",
     last: "http://example.com/articles?page[number]=13&page[size]=1"
  }
```

Reply:
If we go with JSON API we'll have to take a look at http://jsonapi-rb.org/. What if we get this PR out with the current pagination data and discuss it later on so we don't delay this fix?

Reply:
ok :+1: we can do that.
that library doesnt help much with pagination though, we will have to do that part ourselves.